src/xrGame/PHCommander.cpp

```diff
@@ -17,10 +17,10 @@ CPHCall::~CPHCall()
     xr_delete(m_action);
     xr_delete(m_condition);
 }
-bool CPHCall::obsolete() { return !m_action || m_action->obsolete() || !m_condition || m_condition->obsolete(); }
+bool CPHCall::obsolete() { return m_action->obsolete() || m_condition->obsolete(); }
```

Comment:
Are you sure that `m_action` and `m_condition` always aren't null? The previous variant handles the situation when some of these fields is null but yours variant would segfault.

Reply:
I suppose these changes were added as an attempt to fix bugs that have occurred during the adding of tbb allocator.

Reply:
Let's won't remove this check. If you are sure these variables must always be not equal to zero - please add VERIFY or R_ASSERT statement to ensure it. It will help us to discover bugs in the future.